modules/lnd.nix

```diff
@@ -213,6 +213,21 @@ in {
             while ! { exec 3>/dev/tcp/127.0.0.1/${toString cfg.rpcPort}; } &>/dev/null; do
               sleep 0.1
             done
+
+            invoicereadonlymacaroon=${cfg.dataDir}/invoice+readonly.macaroon
```

Comment:
Would be better to add an `invoicereadonlymacaroon` option to lnd.nix which is set by btcpayserver to with path + permissions and this module creates a corresponding macaroon for you. Afaics the `chown lnd` isn't necessary as `lnd` never reads this. As a sidenote this is a bit annoying because just changing your nix config isn't enough to revoke a macaroon.

Reply:
Implemented in 94fcf27

Reply:
sweet!

> As a sidenote this is a bit annoying because just changing your nix config isn't enough to revoke a macaroon.

It actually is enough with your implementation, because the macaroons are created in a temporary directory on every start of the service.